awx/main/utils/licensing.py

```diff
@@ -228,37 +229,42 @@ def validate_rh(self, user, pw):
             host = getattr(settings, 'REDHAT_CANDLEPIN_HOST', None)
 
         if not user:
-            raise ValueError('subscriptions_username is required')
+            raise ValueError('subscriptions_client_id is required')
 
         if not pw:
-            raise ValueError('subscriptions_password is required')
+            raise ValueError('subscriptions_client_secret is required')
 
         if host and user and pw:
             if 'subscription.rhsm.redhat.com' in host:
-                json = self.get_rhsm_subs(host, user, pw)
+                json = self.get_rhsm_subs('https://console.redhat.com/api/rhsm/v2/products?include=providedProducts&oids=480&status=Active', user, pw)
             else:
                 json = self.get_satellite_subs(host, user, pw)
             return self.generate_license_options_from_entitlements(json)
         return []
 
-    def get_rhsm_subs(self, host, user, pw):
-        verify = getattr(settings, 'REDHAT_CANDLEPIN_VERIFY', True)
-        json = []
+    def get_rhsm_subs(self, host, client_id, client_secret):
         try:
-            subs = requests.get('/'.join([host, 'subscription/users/{}/owners'.format(user)]), verify=verify, auth=(user, pw))
+            client = OIDCClient(client_id, client_secret)
+            subs = client.make_request(
+                'GET',
+                host,
+                verify=True,
+                timeout=(31, 31),
+            )
+        except TokenError as e:
+            raise e
         except requests.exceptions.ConnectionError as error:
             raise error
```

Comment:
SonarCloud doesn't like this, and I don't think I do either. A bare `raise` is probably better.

So don't define `error` here. Likewise for `e` in prior except.

Reply:
fixed by just removing this entire try catch, as it's redundant